internal/command/jsonformat/differ/object.go

```diff
@@ -9,36 +9,25 @@ import (
 )
 
 func (v Value) computeAttributeChangeAsObject(attributes map[string]cty.Type) change.Change {
-	var keys []string
-	for key := range attributes {
-		keys = append(keys, key)
-	}
-
-	attributeChanges, changeType := v.processObject(keys, func(key string) interface{} {
-		return attributes[key]
+	attributeChanges, changeType := processObject(v, attributes, func(value Value, ctype cty.Type) change.Change {
+		return value.computeChangeForType(ctype)
 	})
 	return change.New(change.Object(attributeChanges), changeType, v.replacePath())
 }
 
 func (v Value) computeAttributeChangeAsNestedObject(attributes map[string]*jsonprovider.Attribute) change.Change {
-	var keys []string
-	for key := range attributes {
-		keys = append(keys, key)
-	}
-
-	attributeChanges, changeType := v.processObject(keys, func(key string) interface{} {
-		return attributes[key]
+	attributeChanges, changeType := processObject(v, attributes, func(value Value, attribute *jsonprovider.Attribute) change.Change {
+		return value.ComputeChangeForAttribute(attribute)
 	})
 	return change.New(change.NestedObject(attributeChanges), changeType, v.replacePath())
 }
 
-func (v Value) processObject(keys []string, getAttribute func(string) interface{}) (map[string]change.Change, plans.Action) {
+func processObject[T any](v Value, attributes map[string]T, computeChange func(Value, T) change.Change) (map[string]change.Change, plans.Action) {
```

Comment:
A comment here explaining why this must be a generic function instead of a method might be helpful for future readers.

Reply:
Done!